### DISCUSSION TITLE
changing default tasmota_xxxx SSID

### DISCUSSION THREAD

**Discussion by tjkarwal711** [arendst/Tasmota#17924] [Q&A] [answered]
Has anyone changed the default WiFi SSID for this device?

I am trying to have the outlet broadcast a custom SSID rather than the default tasmota_xxxxx when the device boots up for setup. 

Thanks!

**Comment by barbudor** [comment, marked as answer]
As far as I can see the SSID for the SoftAP is equal to the hostname

this is initialized here:
https://github.com/arendst/Tasmota/blob/d029d8bcbcf2391c7e7c58e68871ba7ac2671787/tasmota/tasmota.ino#L603-L608

On a freshly flashed device this comes from the const sting `WIFI_HOSTNAME` which is not overridable:
https://github.com/arendst/Tasmota/blob/d029d8bcbcf2391c7e7c58e68871ba7ac2671787/tasmota/include/tasmota.h#L154

On a configured device, changing hostname and starting `Wificonfig 2` should show an AP named by the hostname
But on a fresh device, I can see a way to override this without changing that line of code
So that's something you would have to report in the code everytime you would merge the base code.

**Reply by sfromis** [reply]
If you want to automatically override host name in a freshly flashed device, you could have your build `user_config_override.h` include a line like:
`#define USER_BACKLOG "Hostname TASHOST; Restart 1"`  (not tested)
Such commands will be executed one time only during fresh initialization, including after a full reset.
